server/apps/microservices/src/processors/metadata-extraction.processor.ts

```diff
@@ -338,3 +341,86 @@ export class MetadataExtractionProcessor {
     return Duration.fromObject({ seconds: videoDurationInSecond }).toFormat('hh:mm:ss.SSS');
   }
 }
+
+@Processor(QueueName.SIDECAR)
+export class SidecarProcessor {
+  private logger = new Logger(SidecarProcessor.name);
+  private assetCore: AssetCore;
+
```

Comment:
The code in `domain/` is basically refactored code, this code here hasn't been updated and doesn't have any tests for it yet. Eventually the thought is to move all the logic out of this file into a `domain/media/media.service.ts` file. If you're up for that, it would be the place to put it. Otherwise, I can move it after it's merged (refactored processors are in `processors.ts` and a new service would need to be exported in `domain.module.ts`). If not, no worries.